#Extraction
######-Loading the raw data for inspection

**Imports and Setup**

In [1]:
from pathlib import Path
import pandas as pd

PROJECT_ROOT = (
    Path.cwd().resolve().parent
    if Path.cwd().resolve().name == 'notebooks'
    else Path.cwd().resolve()
)

**Load Raw Data**

In [4]:
RAW_PATH = PROJECT_ROOT / 'swiggy_file.csv'
df = pd.read_csv(RAW_PATH)
print('Shape:', df.shape)
print('Columns:', df.columns.tolist())
df.head()

Shape: (140657, 10)
Columns: ['Restaurant Name', 'Cuisine', 'Rating', 'Number of Ratings', 'Average Price', 'Number of Offers', 'Offer Name', 'Area', 'Pure Veg', 'Location']


,Restaurant Name,Cuisine,Rating,Number of Ratings,Average Price,Number of Offers,Offer Name,Area,Pure Veg,Location
0,La Pino'Z Pizza,"Pizzas, Pastas",4.0,10+ ratings,₹250 for two,2,FLAT DEAL\nFLAT ₹125 OFF\nUSE FLAT125ABOVE ₹69...,LALA LAJPAT RAI MARKET,No,Abohar
1,The Second Wife,"Indian, North Indian",3.6,50+ ratings,₹250 for two,2,"30% OFF UPTO ₹75\nUSE TRYNEWABOVE ₹149, FLAT ₹...",Central Abohar,No,Abohar
2,Tasty Bites,"Italian, Beverages",3.8,10+ ratings,₹200 for two,1,FLAT ₹120 OFF\nUSE AXIS120ABOVE ₹500,Central Abohar,Yes,Abohar
3,Food Studio,"Pizzas, Burgers",3.5,8 ratings,₹49 for two,5,"50% OFF UPTO ₹100\nUSE TRYNEWABOVE ₹129, FLAT ...",Central Abohar,Yes,Abohar
4,Roll Express,"Fast Food, Snacks",4.3,100+ ratings,₹200 for two,2,DEAL OF DAY\n10% OFF UPTO ₹40\nUSE STEALDEALAB...,Circular Road,No,Abohar


**Structural Inspection**

In [5]:
print('Row count:', len(df))
print('Column count:', len(df.columns))
print(df.dtypes)
print(df.isnull().sum())
print('Duplicates:', df.duplicated().sum())

Row count: 140657
Column count: 10
Restaurant Name      object
Cuisine              object
Rating               object
Number of Ratings    object
Average Price        object
Number of Offers      int64
Offer Name           object
Area                 object
Pure Veg             object
Location             object
dtype: object
Restaurant Name          0
Cuisine                 27
Rating                   0
Number of Ratings    14542
Average Price            0
Number of Offers         0
Offer Name            1808
Area                     2
Pure Veg                 0
Location                 0
dtype: int64
Duplicates: 1336


##The dataset contains 140,657 restaurant records from 581 Indian cities collected in 2024. 10 columns cover restaurant identity, cuisine, pricing, ratings, promotions, location, and dietary preference. Cleaning is required: mixed data types in Rating ('NEW'/'--'), string-formatted prices, 14,542 nulls in Number of Ratings, and 1,336 duplicate rows.